ang/workflow/directives/workflow-list.directive.js

```diff
@@ -55,13 +91,33 @@
      * @returns {Promise} list of workflows
      */
     function getWorkflows (caseTypeCategory) {
-      return civicaseCrmApi('CaseType', 'get', {
-        sequential: 1,
-        case_type_category: caseTypeCategory,
-        options: { limit: 0 }
-      }).then(function (data) {
-        return data.values;
-      });
+      var categoryObject = CaseTypeCategory.findByName(caseTypeCategory);
+      var instanceName = CaseTypeCategory.getCaseTypeCategoryInstance(categoryObject.value).name;
+
+      return getServiceForInstance(instanceName)
+        .getWorkflowsList($scope);
+    }
+
+    /**
+     * Searches for a angularJS service for the current case type category
+     * instance, if not found, returns the service for case management service
+     * as default.
+     *
+     * @param {string} instanceName name of the instance
+     * @returns {object/null} service
+     */
+    function getServiceForInstance (instanceName) {
+      var CASE_MANAGEMENT_INSTACE_NAME = 'case_management';
+
+      try {
+        return $injector.get(
+          CivicaseUtil.capitalizeFirstLetterAndRemoveUnderScore(instanceName) + 'Workflow'
+        );
+      } catch (e) {
+        return $injector.get(
+          CivicaseUtil.capitalizeFirstLetterAndRemoveUnderScore(CASE_MANAGEMENT_INSTACE_NAME) + 'Workflow'
```

Comment:
I think it's better to throw an error if the workflow is not found instead of trying to use a  different service. This would silently fail if the intended service is not found.

Reply:
No no, here is logic is to use the case_management logic as default, if a instance specific logic is not found. So case_management is the default logic here.